### PR TITLE
chore: Added table of contents to help community navigate the examples.

### LINKED ISSUE #269: Provide a table of contents to make it easy to find examples
## Is your feature request related to a problem? Please describe.
We need to provide some guidance on every example application.  A table of contents seems to be the best way.  Each application should provide a description of its intent.

### PR DESCRIPTION
**Pull request by bizob2828**
This PR consolidates some like applications into a parent folder. It also creates a series of READMEs and a root table of contents to help the community navigate the repo easily.

## Links
Closes #269

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,24 @@
 
 # New Relic Node.js Examples
 
-This repository will contain examples of agent features.  Browse the folders for different self-contained, working examples.
+This repository contains example application and scripts that demonstrate functionality of the [New Relic Node.js Agent](https://github.com/newrelic/node-newrelic).
```
Comment (jsumners-nr) [inline README.md:5]:
```suggestion
This repository contains example applications and scripts that demonstrate functionality of the [New Relic Node.js Agent](https://github.com/newrelic/node-newrelic).  
```

File: README.md [hunk 2]
```diff
@@ -2,7 +2,24 @@
 
 # New Relic Node.js Examples
 
-This repository will contain examples of agent features.  Browse the folders for different self-contained, working examples.
+This repository contains example application and scripts that demonstrate functionality of the [New Relic Node.js Agent](https://github.com/newrelic/node-newrelic).  
+
+ * [AI Monitoring](./ai-monitoring) - examples using supported 3rd party libraries to produce AI Monitoring(AIM) telemetry
```
Comment (jsumners-nr) [inline README.md:7]:
```suggestion
 * [AI Monitoring](./ai-monitoring) - examples using supported 3rd party libraries to produce AI Monitoring (AIM) telemetry
```

File: README.md [hunk 3]
```diff
@@ -2,7 +2,24 @@
...
+ * [Logs in Context](./application-logging) - examples using logging libraries to demonstrate agent log decoration and forwarding of logs to New Relic
+ * [Code Level Metrics](./code-level-metrics) - example demonstrating span attributes that help drive the [CodeStream](https://newrelic.com/codestream) integration
+ * [Custom Instrumentation](./custom-instrumentation) - examples demonstrating the use of the [Node.js agent API](https://newrelic.github.io/node-newrelic/API.html)
+ * [Elasticsearch](./elasticsearch) - example demonstrating elasticsearch with Node.js agent
+ * [Error Fingerprinting](./error-fingerprinting) - example using [setErrorCallback](https://newrelic.github.io/node-newrelic/API.html#setErrorGroupCallback) and [noticeError](https://newrelic.github.io/node-newrelic/API.html#noticeError) to [group errors](https://docs.newrelic.com/docs/errors-inbox/errors-inbox/#error-groups) produced in Node.js application
```
Comment (jsumners-nr) [inline README.md:12]:
```suggestion
 * [Error Fingerprinting](./error-fingerprinting) - example using [setErrorCallback](https://newrelic.github.io/node-newrelic/API.html#setErrorGroupCallback) and [noticeError](https://newrelic.github.io/node-newrelic/API.html#noticeError) to [group errors](https://docs.newrelic.com/docs/errors-inbox/errors-inbox/#error-groups) produced in Node.js applications
```

File: README.md [hunk 4]
```diff
@@ -2,7 +2,24 @@
...
+ * [ESM](./esm-app) - example demonstrating how to load agent in an [ESM](https://nodejs.org/api/esm.html) application. It also demonstrates how to register custom instrumentation for a ESM package.
```
Comment (jsumners-nr) [inline README.md:13]:
```suggestion
 * [ESM](./esm-app) - example demonstrating how to load the agent in an [ESM](https://nodejs.org/api/esm.html) application. It also demonstrates how to register custom instrumentation for an ESM package.
```

File: README.md [hunk 5]
```diff
@@ -2,7 +2,24 @@
...
+ * [GraphQL Dataloader](./graphql-koa-dataloader) - example using Apollo Server, koa and GraphQL dataloader
+ * [Kafkajs](./kafkajs) - example demonstrating [kafkajs](https://kafka.js.org/) with Node.js agent
```
Comment (jsumners-nr) [inline README.md:15]:
```suggestion
 * [Kafkajs](./kafkajs) - example demonstrating [kafkajs](https://kafka.js.org/) with the Node.js agent 
```

File: README.md [hunk 6]
```diff
@@ -2,7 +2,24 @@
...
+ * [ESM](./esm-app) - example demonstrating how to load agent in an [ESM](https://nodejs.org/api/esm.html) application. It also demonstrates how to register custom instrumentation for a ESM package.
+ * [GraphQL Dataloader](./graphql-koa-dataloader) - example using Apollo Server, koa and GraphQL dataloader
+ * [Kafkajs](./kafkajs) - example demonstrating [kafkajs](https://kafka.js.org/) with Node.js agent 
+ * [Mock Infinite Tracing Server](./mock-infinite-tracing-server) - mock gRPC server to use to locally test [infinite tracing](https://docs.newrelic.com/docs/distributed-tracing/infinite-tracing/introduction-infinite-tracing/) with a Node.js applciation 
+ * [Nest](./nestjs) - examples demonstrating Nestjs with Node.js agent
+ * [Next.js](./nextjs) - examples demonstrating Next.js with Node.js agent
+ * [Prisma](./prisma-app) - example demonstrating Prisma with Node.js agent
+ * [Sequelize](./sequelize-app) - example demonstrating sequelize with Node.js agent
+ * [Express](./simple-express-app) - example demonstrating express with Node.js agent
```
Comment (jsumners-nr) [inline README.md:21]:
```suggestion
 * [Sequelize](./sequelize-app) - example demonstrating sequelize with the Node.js agent
 * [Express](./simple-express-app) - example demonstrating express with the Node.js agent
```

**Review by jsumners-nr — CHANGES_REQUESTED** [review]

**Review by mrickard — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (mrickard) [reply README.md:21]: I'd recommend the same change with lines 17, 18, and 19 as well.

File: README.md [hunk 7]
```diff
@@ -2,7 +2,24 @@
...
+ * [ESM](./esm-app) - example demonstrating how to load agent in an [ESM](https://nodejs.org/api/esm.html) application. It also demonstrates how to register custom instrumentation for a ESM package.
+ * [GraphQL Dataloader](./graphql-koa-dataloader) - example using Apollo Server, koa and GraphQL dataloader
+ * [Kafkajs](./kafkajs) - example demonstrating [kafkajs](https://kafka.js.org/) with Node.js agent 
+ * [Mock Infinite Tracing Server](./mock-infinite-tracing-server) - mock gRPC server to use to locally test [infinite tracing](https://docs.newrelic.com/docs/distributed-tracing/infinite-tracing/introduction-infinite-tracing/) with a Node.js applciation 
+ * [Nest](./nestjs) - examples demonstrating Nestjs with Node.js agent
+ * [Next.js](./nextjs) - examples demonstrating Next.js with Node.js agent
+ * [Prisma](./prisma-app) - example demonstrating Prisma with Node.js agent
```
Comment (mrickard) [inline README.md:19]:
```suggestion
 * [Nest](./nestjs) - examples demonstrating Nestjs with the Node.js agent
 * [Next.js](./nextjs) - examples demonstrating Next.js with the Node.js agent
 * [Prisma](./prisma-app) - example demonstrating Prisma with the Node.js agent
```

File: custom-instrumentation/README.md [hunk 8]
```diff
@@ -1,2 +1,5 @@
 ## Custom instrumentation
 This folder contains example applications using the [Agent API](https://newrelic.github.io/node-newrelic/API.html) to do custom instrumentation.
+
+ * [instrumentMessages](./instrument-messages) - example application that uses the [newrelic.instrumentMessages](https://newrelic.github.io/node-newrelic/API.html#instrumentMessages) and associated [messaging shim API](https://newrelic.github.io/node-newrelic/MessageShim.html) to instrument a toy messaging library called Nifty Messages
```
Comment (mrickard) [inline custom-instrumentation/README.md:4]:
```suggestion
 * [instrumentMessages](./instrument-messages) - example application that uses the [newrelic.instrumentMessages API](https://newrelic.github.io/node-newrelic/API.html#instrumentMessages) and associated [messaging shim API](https://newrelic.github.io/node-newrelic/MessageShim.html) to instrument a toy messaging library called Nifty Messages
```

File: custom-instrumentation/README.md [hunk 9]
```diff
@@ -1,2 +1,5 @@
...
+ * [instrumentWebframework](./instrument-webframework ) - example application that uses the [newrelic.instrumentWebframework](https://newrelic.github.io/node-newrelic/API.html#instrumentWebframework) and associated [WebFramework shim API](https://newrelic.github.io/node-newrelic/WebFrameworkShim.html) to instrument a hypothetical web framework
```
Comment (mrickard) [inline custom-instrumentation/README.md:5]:
```suggestion
 * [instrumentWebframework](./instrument-webframework ) - example application that uses the [newrelic.instrumentWebframework API](https://newrelic.github.io/node-newrelic/API.html#instrumentWebframework) and associated [WebFramework shim API](https://newrelic.github.io/node-newrelic/WebFrameworkShim.html) to instrument a hypothetical web framework 
```

File: mock-infinite-tracing-server/README.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
+# Mock Infinite Tracing gRPC Server
+
+This is a gRPC server used to test our [infinite tracing](https://docs.newrelic.com/docs/distributed-tracing/infinite-tracing/introduction-infinite-tracing/) with the Node.js agent.
+
+
+## Setup
+
+To setup gRPC run
```
Comment (mrickard) [inline mock-infinite-tracing-server/README.md:8]:
```suggestion
To setup gRPC, run
```

**Review by jsumners-nr — APPROVED** [review]

**Review by mrickard — APPROVED** [review]

**Pull request merged**